dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/context/AbstractSupplier.java

```diff
@@ -27,17 +27,75 @@
  */
 package org.hisp.dhis.dxf2.events.importer.context;
 
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hisp.dhis.commons.jackson.config.JacksonObjectMapperConfig;
+import org.hisp.dhis.commons.util.SystemUtils;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKTReader;
+import org.springframework.core.env.Environment;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+
 /**
  * @author Luciano Fiandesio
  */
-public abstract class AbstractSupplier<T> implements WorkContextSupplier<T>
+@Slf4j
+public abstract class AbstractSupplier
 {
     protected final NamedParameterJdbcTemplate jdbcTemplate;
 
-    public AbstractSupplier( NamedParameterJdbcTemplate jdbcTemplate )
+    protected final Environment environment;
+
+    public AbstractSupplier( NamedParameterJdbcTemplate jdbcTemplate, Environment environment )
     {
         this.jdbcTemplate = jdbcTemplate;
+        this.environment = environment;
+    }
+
+    public <T> T jsonToEntity( String json, Class<T> clazz )
+    {
+        if ( StringUtils.isEmpty( json ) )
+            return null;
+
+        try
+        {
+            return JacksonObjectMapperConfig.staticJsonMapper().readValue( json, clazz );
+        }
+        catch ( JsonProcessingException e )
+        {
+            log.error( e.getMessage() );
+        }
+
+        return null;
+    }
+
+    public Geometry getGeometryFrom( String field, ResultSet rs, String entity )
+        throws SQLException
+    {
+        if ( null != rs.getObject( field ) )
+        {
+            try
+            {
+                return new WKTReader().read( rs.getString( field ) );
+            }
+            catch ( ParseException e )
+            {
+                log.error( "Unable to read geometry for '" + entity + "': ", e );
+            }
+        }
+
+        return null;
+    }
+
+    public String getGeometryField( String field )
+    {
+        return SystemUtils.isH2( environment.getActiveProfiles() ) ? field : "ST_AsText( " + field + " )";
```

Comment:
Can we try to avoid checking at runtime if this is a test or not?
We were trying to walk away from that as it is causing some performance issues in other parts of the system

Reply:
I don't see how this is creating performance issues as is env variable check, in any case, the only way I can think of to improve it is to remove H2 tests or convert it to a not-h2 test